service/src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java

```diff
@@ -475,8 +473,12 @@ public ResponseEntity<Void> grantRole(
     // No additional authz check as this is just a wrapper around a Sam endpoint.
     SamRethrow.onInterrupted(
         () ->
-            samService.grantWorkspaceRole(
-                uuid, getAuthenticatedInfo(), WsmIamRole.fromApiModel(role), body.getMemberEmail()),
+            getSamService()
```

Comment:
Why do we need to getSamService() if it is AutoWired?

Reply:
Autowire annotation is used for Dependency Injection -> it injects required beans (samService) to the current bean (WorkspaceApiController) for its construction.

In other words, this annotation provides the objects as constructor args at line 113 (note: this constructor is not manually called anywhere and changing the arg order does not cause an error).

After construction the external beans are not available and a ref needs to be stored during construction if needed. The parent class has this field as a private member, and hence need the getter to access it in the child class